src/components/ObsEdit/EvidenceSection.js

```diff
@@ -91,6 +95,19 @@ const EvidenceSection = ( {
         positional_accuracy: location?.positional_accuracy
       } );
 
+      // Update all photos taken via the app with the new fetched location.
+      // We wrap this in a setTimeout since sometimes the location gets returned quickly, before
+      // the background code in ObsEditProvider finished writing the photos into the camera roll
+      // (and in this case, we don't mind updating the photos a little later, since it's all done
+      // in the background anyhow).
+      if ( location ) {
+        setTimeout( () => {
+          cameraRollUrisRef.current.forEach( uri => {
+            writeExifToFile( uri, { latitude: location.latitude, longitude: location.longitude } );
+          } );
+        }, 3000 );
```

Comment:
1. I don't love using `setTimeout` here. If you need to wait until a file is written to a specific URI, why not track the the state of write success and only try to write the EXIF when you know the file exists? That would resolve the race condition in a more deterministic way
2. I don't think this component needs to know about `cameraRollUris`. Maybe ObsEditProvider should export a method like `updateCameraRollPhotosWithExif()` or something?